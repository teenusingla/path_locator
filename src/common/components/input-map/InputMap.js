import React, { Component } from "react";
import { maps } from "../../../common/services";
import "./InputMap.scss";

class InputMap extends Component {
  //Autocomplete, After rendering the google autocomplete, reference will be saved in this
  inputMapCtrl;

  constructor(props) {
    super(props);
    this.inputMapRef = React.createRef();
  }

  /**
   * @name renderAutoComplete
   * @description Attach the google places autocomplete to the inputs
   */
  renderAutoComplete = async () => {
    const maps = await this.props.maps();

    this.inputMapCtrl = new maps.places.Autocomplete(this.inputMapRef.current);

    // Adding the eventlistener for place_change to google Autocomplete
    this.inputMapCtrl.addListener("place_changed", e => {
      const places = this.inputMapCtrl.getPlace();
      if (places) {
        this.props.mapValueChange({ name: this.props.name, value: places });
      }
    });
  };

  /**
   * Component lifecycle hook
   */
  componentDidMount() {
    this.renderAutoComplete();
  }

  componentDidUpdate() {
    if (!this.props.value) {
      this.inputMapRef.current.value = "";
    }
  }

  // Method for detecting empty values and updating the props onChange event
  onChange = e => {
    if (e.target.value.trim() === "") {
      this.props.mapValueChange({ name: this.props.name, value: null });
    }
  };

  render() {
    return (
      <div className="rows">
        <p>{this.props.label}</p>
        <input
          className="input-map"
          type="text"
          ref={this.inputMapRef}
          name={this.props.name}
          onChange={this.onChange}
        />
      </div>
    );
  }
}

InputMap.defaultProps = {
  maps
};

export default InputMap;
