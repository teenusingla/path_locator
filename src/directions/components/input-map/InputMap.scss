@import "../../../responsive.scss";

.error {
  font-size: 12px;
  color: #ee3124;
  margin-top: 7px;
}

.input-map {
  height: 25px;
  width: 210px;
}
