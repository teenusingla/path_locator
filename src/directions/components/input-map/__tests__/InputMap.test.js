import React from "react";
import renderer from "react-test-renderer";
import InputMap from "../InputMap";

describe("<InputMap /> component", () => {
  it("should match the snapshot", () => {
    const tree = renderer.create(<InputMap />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
