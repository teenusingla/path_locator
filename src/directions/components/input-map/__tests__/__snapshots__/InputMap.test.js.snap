// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputMap /> component should match the snapshot 1`] = `
<div
  className="rows"
>
  <p />
  <input
    className="input-map"
    onChange={[Function]}
    type="text"
  />
</div>
`;
